install.sh

```diff
@@ -70,25 +76,27 @@ detect_arch() {
   printf '%s' "${arch}"
 }
 
-platform="$(detect_platform)"
-arch="$(detect_arch)" || abort "Sorry! pnpm currently only provides pre-built binaries for x86_64/arm64 architectures."
-pkgName="@pnpm/${platform}-${arch}"
-version_json="$(download "https://registry.npmjs.org/${pkgName}")" || abort "Download Error!"
-version="$(printf '%s' "${version_json}" | tr '{' '\n' | awk -F '"' '/latest/ { print $4 }')"
-archive_url="https://registry.npmjs.org/${pkgName}/-/${platform}-${arch}-${version}.tgz"
```

Comment:
move var related `download_and_install` function inside the function for better readability

Reply:
@oplik0 has added the functions I believe. @oplik0 are you fine with these changes?

Reply:
The function was added by @umireon in https://github.com/pnpm/get.pnpm.io/commit/ed4adf7e230b0cbb2a606696eeba5f42dc285413 and the variables have been here from the start - you might be remembering me here because you added them to `beta-install.sh` to the PR I made (my part of which was adding a case for aarch64): https://github.com/pnpm/get.pnpm.io/pull/14/commits/7831416cf8d748207a1fac6cd8f85b08c1d9eb34

My contribution to this repository was much more minor :)

But it does look better in my opinion.

Reply:
I just respected the signature of the old shell function.
I have no strong opinion about the place of these variables and the current place looks better.